lib/src/html/html_generator.dart

```diff
@@ -50,33 +44,40 @@ class HtmlGenerator extends Generator {
 
   /// [url] - optional URL for where the docs will be hosted.
   static Future<HtmlGenerator> create(
-      {String url,
+      {HtmlGeneratorOptions options,
       List<String> headers,
-      List<String> footers,
-      String relCanonicalPrefix,
-      String toolVersion,
-      String faviconPath,
-      bool useCategories: false}) async {
+      List<String> footers}) async {
     var templates =
         await Templates.create(headerPaths: headers, footerPaths: footers);
 
-    if (toolVersion == null) {
-      toolVersion = 'unknown';
-    }
-
-    return new HtmlGenerator._(url, relCanonicalPrefix, templates, toolVersion,
-        faviconPath: faviconPath, useCategories: useCategories);
+    return new HtmlGenerator._(
+        options ?? new HtmlGeneratorOptions(), templates);
   }
 
-  HtmlGenerator._(
-      this._url, this._relCanonicalPrefix, this._templates, this._toolVersion,
-      {this.faviconPath, this.useCategories});
+  HtmlGenerator._(this._options, this._templates);
 
   @override
   Future generate(Package package, Directory out) {
-    return new HtmlGeneratorInstance(_toolVersion, _url, _templates, package,
-            out, _onFileCreated, _relCanonicalPrefix,
-            faviconPath: faviconPath, useCategories: useCategories)
+    return new HtmlGeneratorInstance(
+            _options, _templates, package, out, _onFileCreated)
         .generate();
   }
 }
+
+class HtmlGeneratorOptions {
```

Comment:
This looks better now, I can go with this; it does make things a bit neater in HtmlGenerator and its implementation.